pkg/clients/aws.go

```diff
@@ -85,12 +104,21 @@ func GetConfig(ctx context.Context, c client.Client, mg resource.Managed, region
 	}
 }
 
-// UseProviderConfig to produce a config that can be used to authenticate to AWS.
-func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed, region string) (*aws.Config, error) {
+// GetProviderConfig retrieves the providerConfig from kubernetes associated with a managed resource
+func GetProviderConfig(ctx context.Context, c client.Client, mg resource.Managed) (*v1beta1.ProviderConfig, error) {
 	pc := &v1beta1.ProviderConfig{}
 	if err := c.Get(ctx, types.NamespacedName{Name: mg.GetProviderConfigReference().Name}, pc); err != nil {
 		return nil, errors.Wrap(err, "cannot get referenced Provider")
 	}
+	return pc, nil
+}
+
+// UseProviderConfig to produce a config that can be used to authenticate to AWS.
+func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed, region string) (*aws.Config, error) {
+	pc, err := GetProviderConfig(ctx, c, mg)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
	pc := &v1beta1.ProviderConfig{}
	if err := c.Get(ctx, types.NamespacedName{Name: mg.GetProviderConfigReference().Name}, pc); err != nil {
		return nil, errors.Wrap(err, "cannot get referenced Provider")
	}
	switch {
	case region == "" && pc.Spec.Credentials.Partition != "":
		region = pc.Spec.Credentials.Partition
	case region == "" && pc.Spec.Credentials.Partition == "":
		region = "aws-global"
	}
```

Would we be able to get away with something like this by letting individual controllers pass in `""` as region if they don't have that field?